samples/simpleasset/simpleasset.go

```diff
@@ -0,0 +1,238 @@
+	}
+	if !exists {
+		return fmt.Errorf("the asset %s does not exist", id)
+	}
+
+	return ctx.GetStub().DelState(id)
+}
+
+// AssetExists returns true when asset with given ID exists in world state
+func (s *SmartContract) isAssetExists(ctx contractapi.TransactionContextInterface, id string) (bool, error) {
+	assetJSON, err := ctx.GetStub().GetState(id)
+	if err != nil {
+		return false, fmt.Errorf("failed to read from world state: %v", err)
+	}
+
+	return assetJSON != nil, nil
+}
+// AssetExists returns true when asset with given ID exists in world state
+func (s *SmartContract) isAssetReleased(ctx contractapi.TransactionContextInterface, id string) (bool, error) {
+	asset, err := s.ReadAsset(ctx, id)
+	if err != nil {
+		return false, err
+	}
+	currDate := time.Now()
+	if (currDate.After(asset.MaturityDate)) {
+		return true, nil
+	}
+
+	return false, nil
+}
+
+// TransferAsset updates the owner field of asset with given id in world state.
+func (s *SmartContract) UpdateOwner(ctx contractapi.TransactionContextInterface, id string, newOwner string) error {
+	asset, err := s.ReadAsset(ctx, id)
+	if err != nil {
+		return err
+	}
+
+	asset.Owner = newOwner
+	assetJSON, err := json.Marshal(asset)
+	if err != nil {
+		return err
+	}
+
+	return ctx.GetStub().PutState(id, assetJSON)
+}
+// UpdateAsset updates an existing asset in the world state with provided parameters.
+func (s *SmartContract) UpdateMaturityDate(ctx contractapi.TransactionContextInterface, id string, newMaturityDate time.Time) error {
+	asset, err := s.ReadAsset(ctx, id)
+	if err != nil {
+		return err
+	}
+
+	asset.MaturityDate = newMaturityDate
+	assetJSON, err := json.Marshal(asset)
+	if err != nil {
+		return err
+	}
+
+	return ctx.GetStub().PutState(id, assetJSON)
+}
+// UpdateAsset updates an existing asset in the world state with provided parameters.
+func (s *SmartContract) UpdateFaceValue(ctx contractapi.TransactionContextInterface, id string, newFaceValue int) error {
+	asset, err := s.ReadAsset(ctx, id)
+	if err != nil {
+		return err
+	}
+
+	asset.FaceValue = newFaceValue
+	assetJSON, err := json.Marshal(asset)
+	if err != nil {
+		return err
+	}
+
+	return ctx.GetStub().PutState(id, assetJSON)
+}
+
+// GetAllAssets returns all assets found in world state
+func (s *SmartContract) GetAllAssets(ctx contractapi.TransactionContextInterface) ([]*Asset, error) {
+	// range query with empty string for startKey and endKey does an
+	// open-ended query of all assets in the chaincode namespace.
+	resultsIterator, err := ctx.GetStub().GetStateByRange("", "")
+	if err != nil {
+		return nil, err
+	}
+	defer resultsIterator.Close()
+
+	var assets []*Asset
+	for resultsIterator.HasNext() {
+		queryResponse, err := resultsIterator.Next()
+		if err != nil {
+			return nil, err
+		}
+
+		var asset Asset
+		err = json.Unmarshal(queryResponse.Value, &asset)
+		if err != nil {
+			return nil, err
+		}
+		assets = append(assets, &asset)
+	}
+
+	return assets, nil
+}
+
+func main() {
```

Comment:
This chaincode serves the purpose for a non-fungible asset. But we'd like to support one kind of fungible asset as well, say a `Token` or a `DigitalCash` (to be more explicit). Do you want to add code for that in this PR or in another one?

Reply:
I'll add that code in next PR. Do you suggest a separate chaincode for that? because the functions might behave differently for a fungible asset, so I was thinking it to be a different chaincode.